src/DevBetterWeb.Infrastructure/Handlers/DailyCheckInitiatedEventHandler.cs

```diff
@@ -29,13 +38,24 @@ public async Task Handle(DailyCheckInitiatedEvent domainEvent)
       await _alumniGraduationService.GraduateMembersIfNeeded(messages);
 
       // check if people need to be pinged about new member link
+      await _dailyCheckPingService.SendPingIfNeeded(messages);
 
       messages.Append("Daily Check Event Completed");
 
+      var spec = new DailyCheckByDateSpec(DateTime.Today);
+
+      var todaysDailyCheck = await _repository.GetBySpecAsync(spec);
+
       foreach (var message in messages)
       {
         _webhook.Content = message;
         await _webhook.Send();
+
+        if(todaysDailyCheck != null && !message.Equals("Daily Check Event Completed"))
+        {
+          if (todaysDailyCheck.TasksCompleted == null) todaysDailyCheck.TasksCompleted = "";
+          todaysDailyCheck.TasksCompleted += message;
+        }
```

Comment:
This seems overly complicated and reliant upon magic strings, which you're duplicating on lines 43 and 54. Is there a way to simplify this logic?

Reply:
do you want to have to store "daily check event completed" in the db for every day, or can I omit that? That's basically all this is trying to accomplish. But I'll at least pull some of it into its own method so it's a bit more clear

Reply:
no need to store that in the db every day, no. no need to keep/persist a log of every check - the discord history will do a good enough job of that.